File: CS/EnumItemsSourceBehaviorExample/Common/FilterStringConverter.cs
using System;
using System.Linq;
using System.Windows.Data;
using System.Collections.Generic;

namespace EnumItemsSourceBehaviorExample.Common {
    public class FilterStringConverter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            if (value == null) return Binding.DoNothing;
            return String.Format("[{0}] = '{1}'", parameter, value);
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            throw new NotImplementedException();
        }
    }
}

File: VB/EnumItemsSourceBehaviorExample/Common/FilterStringConverter.vb
Imports System
Imports System.Windows.Data

Namespace EnumItemsSourceBehaviorExample.Common

    Public Class FilterStringConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.Convert
            If value Is Nothing Then Return Binding.DoNothing
            Return String.Format("[{0}] = '{1}'", parameter, value)
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace
